python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,346 @@
+##########################################################################
+# Ganga Project. http://cern.ch/ganga
+#
+# $Id: Executable.py,v 1.1 2008-07-17 16:40:57 moscicki Exp $
+##########################################################################
+
+from Ganga.GPIDev.Adapters.IPrepareApp import IPrepareApp
+from Ganga.GPIDev.Adapters.IRuntimeHandler import IRuntimeHandler
+from Ganga.GPIDev.Schema import Schema, Version, SimpleItem
+
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.File import File, ShareDir
+from Ganga.GPIDev.Lib.File.FileBuffer import FileBuffer
+from Ganga.Core import ApplicationConfigurationError, ApplicationPrepareError
+
+from Ganga.Utility.logging import getLogger
+
+from Ganga.GPIDev.Base.Proxy import getName
+
+import os
+import shutil
+from datetime import datetime
+import tempfile
+import tarfile
+import random
+import threading
+from Ganga.Utility.files import expandfilename
+
+from Ganga.GPIDev.Adapters.ApplicationRuntimeHandlers import allHandlers
+
+from GangaGaudi.Lib.RTHandlers.RunTimeHandlerUtils import get_share_path, master_sandbox_prepare, sandbox_prepare, script_generator
+from GangaDirac.Lib.RTHandlers.DiracRTHUtils import dirac_inputdata, dirac_ouputdata, mangle_job_name, diracAPI_script_template, diracAPI_script_settings, API_nullifier, dirac_outputfile_jdl
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+from Ganga.GPIDev.Lib.File.OutputFileManager import getOutputSandboxPatterns, getWNCodeForOutputPostprocessing
+from Ganga.GPIDev.Adapters.StandardJobConfig import StandardJobConfig
+from Ganga.Utility.util import unique
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+
+logger = getLogger()
+
+data_file = 'data.py'
+
+prep_lock = threading.Lock()
```

Comment:
Yes if this is to enable multi-threaded submit. This is currently due to a bug in Job which hasn't been fixed yet.